libsigner/src/messages.rs

```diff
@@ -168,35 +219,135 @@ impl From<&RejectCode> for RejectCodeTypePrefix {
             },
-            Self::BlockResponse(_) => BLOCK_MSG_ID,
-            Self::Transactions(_) => TRANSACTIONS_MSG_ID,
+            Self::BlockResponse(_) => MessageSlotID::BlockResponse,
+            Self::Transactions(_) => MessageSlotID::Transactions,
+            Self::DkgResults { .. } => MessageSlotID::DkgResults,
+        }
+    }
+}
+
+impl SignerMessage {
+    /// Provide an interface for consensus serializing a DkgResults `SignerMessage`
+    ///  without constructing the DkgResults struct (this eliminates a clone)
+    pub fn serialize_dkg_result<'a, W: Write, I>(
+        fd: &mut W,
+        aggregate_key: &Point,
+        party_polynomials: I,
+    ) -> Result<(), CodecError>
+    where
+        I: ExactSizeIterator + Iterator<Item = (&'a u32, &'a PolyCommitment)>,
+    {
+        SignerMessageTypePrefix::DkgResults
+            .to_u8()
+            .consensus_serialize(fd)?;
+        Self::serialize_dkg_result_components(fd, aggregate_key, party_polynomials)
+    }
+
+    /// Serialize the internal components of DkgResults (this eliminates a clone)
+    fn serialize_dkg_result_components<'a, W: Write, I>(
+        fd: &mut W,
+        aggregate_key: &Point,
+        party_polynomials: I,
+    ) -> Result<(), CodecError>
+    where
+        I: ExactSizeIterator + Iterator<Item = (&'a u32, &'a PolyCommitment)>,
+    {
+        fd.write_all(&aggregate_key.compress().data)
+            .map_err(CodecError::WriteError)?;
+        let polynomials_len: u32 = party_polynomials
+            .len()
+            .try_into()
+            .map_err(|_| CodecError::ArrayTooLong)?;
+        polynomials_len.consensus_serialize(fd)?;
+        for (party_id, polynomial) in party_polynomials {
+            party_id.consensus_serialize(fd)?;
+            fd.write_all(&polynomial.id.id.to_bytes())
+                .map_err(CodecError::WriteError)?;
+            fd.write_all(&polynomial.id.kG.compress().data)
+                .map_err(CodecError::WriteError)?;
+            fd.write_all(&polynomial.id.kca.to_bytes())
+                .map_err(CodecError::WriteError)?;
+            let commit_len: u32 = polynomial
+                .poly
+                .len()
+                .try_into()
+                .map_err(|_| CodecError::ArrayTooLong)?;
+            commit_len.consensus_serialize(fd)?;
+            for poly in polynomial.poly.iter() {
+                fd.write_all(&poly.compress().data)
+                    .map_err(CodecError::WriteError)?;
+            }
         }
+        Ok(())
+    }
+
+    fn deserialize_point<R: Read>(fd: &mut R) -> Result<Point, CodecError> {
+        let mut bytes = [0; 33];
+        fd.read_exact(&mut bytes).map_err(CodecError::ReadError)?;
+        Point::try_from(&Compressed::from(bytes))
+            .map_err(|e| CodecError::DeserializeError(e.to_string()))
+    }
+
+    fn deserialize_scalar<R: Read>(fd: &mut R) -> Result<Scalar, CodecError> {
```

Comment:
We already have an implimentation of this no?

StacksMessageCodecExtensions is implemented for Scalar and for Point?

Reply:
Can we delete deserialize_scalar and deserialize_point then if they are no longer used?